File: app/src/main/java/com/lifedawn/bestweather/room/dao/DailyPushNotificationDao.java
package com.lifedawn.bestweather.room.dao;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import com.lifedawn.bestweather.room.dto.AlarmDto;
import com.lifedawn.bestweather.room.dto.DailyPushNotificationDto;

import java.util.List;

@Dao
public interface DailyPushNotificationDao {
	@Query("SELECT * FROM daily_push_notifications_table")
	List<DailyPushNotificationDto> getAll();

	@Query("SELECT * FROM daily_push_notifications_table")
	LiveData<List<DailyPushNotificationDto>> list();

	@Query("SELECT count(*) FROM daily_push_notifications_table")
	int size();

	@Query("SELECT * FROM daily_push_notifications_table WHERE id = :id")
	DailyPushNotificationDto get(int id);

	@Delete
	void delete(DailyPushNotificationDto dailyPushNotificationDto);

	@Insert(entity = DailyPushNotificationDto.class)
	long add(DailyPushNotificationDto dailyPushNotificationDto);

	@Update(onConflict = OnConflictStrategy.IGNORE)
	int update(DailyPushNotificationDto dailyPushNotificationDto);
}

File: app/src/main/java/com/lifedawn/bestweather/data/local/room/dao/DailyPushNotificationDao.kt
package com.lifedawn.bestweather.data.local.room.dao

import androidx.room.*
import com.lifedawn.bestweather.data.local.room.dto.DailyPushNotificationDto
import kotlinx.coroutines.flow.Flow

@Dao
interface DailyPushNotificationDao {
    @get:Query("SELECT * FROM daily_push_notifications_table") val all: Flow<List<DailyPushNotificationDto>>

    @Query("SELECT * FROM daily_push_notifications_table")
    fun list(): Flow<List<DailyPushNotificationDto>>

    @Query("SELECT count(*) FROM daily_push_notifications_table")
    fun size(): Flow<Int>

    @Query("SELECT * FROM daily_push_notifications_table WHERE id = :id")
    operator fun get(id: Int): Flow<DailyPushNotificationDto>

    @Delete
    fun delete(dailyPushNotificationDto: DailyPushNotificationDto)

    @Insert(entity = DailyPushNotificationDto::class)
    fun add(dailyPushNotificationDto: DailyPushNotificationDto): Flow<Long>

    @Update(onConflict = OnConflictStrategy.IGNORE)
    fun update(dailyPushNotificationDto: DailyPushNotificationDto): Flow<Int>
}